### ISSUE TITLE
Panic: runtime error: index out of range when issuing query

### ISSUE THREAD

**Issue by nemothekid** [influxdb/influxdb#2396]
**Environment**: ubuntu 14.04 64-bit running Docker which installs https://s3.amazonaws.com/influxdb/influxdb_0.9.0-rc23_amd64.deb 

Steps 
1.) Created a `collectd` database
2.) Setup collectd to push to influxdb
3.) Ran query - 
```
[http] 2015/04/22 19:57:44 172.17.42.1 - - [22/Apr/2015:19:57:44 +0000] GET /query?db=collectd&p=root&q=SELECT+mean(df_used)+FROM+%22df_used%22+WHERE+time+%3E+now()+-+6h+AND+host%3D%27cass1.channelmeter.com%27+AND+type_instance%3D%27root%27+GROUP+BY+time(30s)+ORDER+BY+asc&u=root HTTP/1.1 200 5691 http://metrics.marathon.mesos/dashboard/db/cassandra?panelId=2&fullscreen&edit Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36 d7a39f9a-e929-11e4-8006-000000000000 6.524426ms
```

Panic Stack Trace:
```
panic: runtime error: index out of range

goroutine 2926 [running]:
github.com/influxdb/influxdb/influxql.InitializeReduceFunc(0xc2080dcb10, 0x1, 0x0, 0x0)
	/home/philip/build/src/github.com/influxdb/influxdb/influxql/functions.go:110 +0x62b
github.com/influxdb/influxdb/influxql.(*MapReduceJob).Execute(0xc20818e850, 0xc208525bc0, 0x0)
	/home/philip/build/src/github.com/influxdb/influxdb/influxql/engine.go:89 +0x274
github.com/influxdb/influxdb/influxql.(*Executor).execute(0xc208579800, 0xc208525bc0)
	/home/philip/build/src/github.com/influxdb/influxdb/influxql/engine.go:771 +0xba
created by github.com/influxdb/influxdb/influxql.(*Executor).Execute
	/home/philip/build/src/github.com/influxdb/influxdb/influxql/engine.go:750 +0x5a
```

**Comment by toddboom** [comment]
@nemothekid would you be able to try this with a newer version of InfluxDB? we're just about to cut rc27, and it's possible that this has been fixed in a more recent version.

**Comment by nemothekid** [comment]
With 0.9.0-rc26 - same query crashes w/

```
panic: unsupported value type during decode by id: influxql.DataType

goroutine 13262 [running]:
github.com/influxdb/influxdb.(*FieldCodec).DecodeByID(0xc2080ce2a0, 0x1, 0x7fafb0fe673a, 0x9, 0x7ffffbf5, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/database.go:897 +0x66b
github.com/influxdb/influxdb.(*LocalMapper).Next(0xc20829da20, 0x4e, 0x13d75fda16982200, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tx.go:463 +0x6b2
github.com/influxdb/influxdb/influxql.MapMean(0x7fafbd383060, 0xc20829da20, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/functions.go:212 +0x69
github.com/influxdb/influxdb.(*LocalMapper).NextInterval(0xc20829da20, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tx.go:401 +0xd3
github.com/influxdb/influxdb/influxql.(*MapReduceJob).processAggregate(0xc2082beaf0, 0xc2081b6630, 0xac2db8, 0xc208282000, 0x2d1, 0x2d1, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:626 +0x264
github.com/influxdb/influxdb/influxql.(*MapReduceJob).Execute(0xc2082beaf0, 0xc2081ff8c0, 0xc2080e5f00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:167 +0x8bf
github.com/influxdb/influxdb/influxql.(*Executor).execute(0xc2082b09c0, 0xc2081ff8c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:774 +0xba
created by github.com/influxdb/influxdb/influxql.(*Executor).Execute
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:753 +0x5a
```

**Comment by jwilder** [comment]
First panic should be fixed via #2404
